src/pages/StocktakeEditPage.js

```diff
@@ -213,16 +233,16 @@ export const StocktakeEditPage = ({
         renderHeader={renderHeader}
         keyExtractor={keyExtractor}
         getItemLayout={getItemLayout}
+        columns={columns}
       />
       <DataTablePageModal
         fullScreen={false}
         isOpen={!!modalKey}
         modalKey={modalKey}
-        onClose={() => dispatch(closeBasicModal())}
+        onClose={modalKey === MODAL_KEYS.EDIT_STOCKTAKE_BATCH ? onConfirmBatchEdit : onCloseModal}
```

Comment:
possibly a `useSelectOnClose` prop which would use the `onSelect` method as the parent `ModalContainers` `onClose` prop? Seems like a relatively common use case